### PR TITLE
add baritone pause

### PR DESCRIPTION
**Pull request by LordMZTE**

### CODE REVIEW

File: src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritonePause.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        process?.process?.isPaused = false
+    }
+
+    override fun onEnable() {
+        process?.process?.isPaused = true
+    }
+}
+
+// this wrapper is required in order to keep kami from crashing without baritone
+class BaritoneData(val process: PauseBaritoneProcess) {
```
Comment (5HT2) [inline src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritonePause.kt]: Have a better name, such as BaritoneWrapper

File: src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritonePause.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    class PauseBaritoneProcess(
+        val displayName: String,
+        var isPaused: Boolean = false
+    ) : IBaritoneProcess {
+        override fun isActive(): Boolean = isPaused
+
+        override fun onTick(p0: Boolean, p1: Boolean) =
+            PathingCommand(null, PathingCommandType.REQUEST_PAUSE)
```
Comment (5HT2) [inline src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritonePause.kt:59]: why are you doing this???

**Review by 5HT2 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritonePause.kt (hunk 2 above)
Comment (LordMZTE) [reply src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritonePause.kt:59]: because baritone's pause command does this

Comment (5HT2) [reply src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritonePause.kt:59]: > https://github.com/kami-blue/client/blob/master/src/main/java/me/zeroeightsix/kami/process/TemporaryPauseProcess.kt

Nevermind, you're right

Comment (LordMZTE) [reply src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritonePause.kt:59]: also onTick needs to return a PathingCommand

File: src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritonePause.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    private fun tryInitProcess() {
+        if (registeredProcess) return
+        BaritoneIntegration {
+            val mngr = BaritoneAPI.getProvider()?.primaryBaritone?.pathingControlManager
+            mngr?.let {
+                it.registerProcess(process?.process)
+                registeredProcess = true
+            }
+        }
+    }
```
Comment (zeroeightysix) [inline src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritonePause.kt]: This doesn't sit right with me. It's rather yucky to
1. maintain state with `registeredProcess`
2. have a tick listener that will basically do nothing except for 1 tick

I'd rather have you subscribe a one-off listener that unsubscribes itself:

```kt
BaritoneIntegration {
    hidden = false
    process = BaritoneWrapper(BaritoneWrapper.PauseBaritoneProcess("${KamiMod.MODNAME} Pause Module"))
    var listener: Listener<TickEvent.InGame>? = null
    listener = Listener({
        val manager = BaritoneAPI.getProvider()?.primaryBaritone?.pathingControlManager
        manager?.let {
            it.registerProcess(process?.process)
            KamiMod.EVENT_BUS.unsubscribe(listener)
        }
    })
    KamiMod.EVENT_BUS.subscribe(listener)
}
```
This way there is no need for the tick listener or `tryInitProcess`

**Review by zeroeightysix — CHANGES_REQUESTED** [review]

**Review by zeroeightysix — APPROVED** [review]

**Comment by zeroeightysix** [comment]
Thanks!

**Pull request merged**